cmd/cluster/resizecontrolplanenode.go

```diff
@@ -0,0 +1,301 @@
+
+	err = (*awsClient).WaitUntilInstanceRunning(describeInstancesInput)
+	if err != nil {
+		log.Fatalln("Unable to get ec2 instance up and running", err)
+	}
+}
+
+func uncordonNode(nodeID string) {
+	fmt.Println("Uncordoning node", nodeID)
+
+	// TODO: replace subprocess call with API call
+	cmd := fmt.Sprintf("oc adm uncordon %s", nodeID)
+	_, err := exec.Command("bash", "-c", cmd).Output()
+	if err != nil {
+		log.Fatalln(err)
+	}
+}
+
+// Start and stop calls require the internal AWS instance ID
+// Machinetype patch requires the tag "Name"
+func getNodeAwsInstanceData(node string, awsClient *awsprovider.Client) (string, string) {
+	params := &ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("private-dns-name"),
+				Values: []*string{aws.String(node)},
+			},
+		},
+	}
+	ret, err := (*awsClient).DescribeInstances(params)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	awsInstanceID := *(ret.Reservations[0].Instances[0].InstanceId)
+
+	var machineName string = ""
+	tags := ret.Reservations[0].Instances[0].Tags
+	for _, t := range tags {
+		if *t.Key == "Name" {
+			machineName = *t.Value
+		}
+	}
+
+	if machineName == "" {
+		log.Fatalln("Could not retrieve node machine name.")
+	}
+
+	fmt.Println("Node", node, "found as AWS internal InstanceId", awsInstanceID, "with machine name", machineName)
+
+	return machineName, awsInstanceID
+}
+
+func patchMachineType(machine string, machineType string) {
+	fmt.Println("Patching machine type of machine", machine, "to", machineType)
+	cmd := `oc -n openshift-machine-api patch machine ` + machine + ` --patch "{\"spec\":{\"providerSpec\":{\"value\":{\"instanceType\":\"` + machineType + `\"}}}}" --type merge --as backplane-cluster-admin`
+	err := exec.Command("bash", "-c", cmd).Run()
+	if err != nil {
+		log.Fatalln("Could not patch machine type:", err)
+	}
+}
+
+func (o *resizeControlPlaneNodeOptions) run() error {
+	awsClient, err := clustercloud.CreateAWSClient(o.clusterID)
+	if err != nil {
+		return err
+	}
+
+	machineName, nodeAwsID := getNodeAwsInstanceData(o.node, &awsClient)
+
+	// drain node with oc adm drain <node> --ignore-daemonsets --delete-emptydir-data
+	drainNode(o.node)
+
+	// Stop the node instance
+	stopNode(&awsClient, nodeAwsID)
+
+	// Once stopped, change the instance type
+	modifyInstanceAttribute(&awsClient, nodeAwsID, o.newMachineType)
+
+	// Start the node instance
+	startNode(&awsClient, nodeAwsID)
+
+	// uncordon node with oc adm uncordon <node>
+	uncordonNode(o.node)
+
+	fmt.Println("To continue, please confirm that the node is up and running and that the cluster is in the desired state to proceed.")
+	err = utils.ConfirmSend()
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("To finish the node resize, it is suggested to update the machine spec. This requires ***elevated privileges***. Do you want to proceed?")
+	err = utils.ConfirmSend()
+	if err != nil {
+		fmt.Println("Node resized, machine type not patched. Exiting...")
+		return nil
```

Comment:
Why is the error not returned here?

Reply:
Hmm. I believe this is a leftover from when `ConfirmSend` returned the boolean and I forgot to change it to return the error. 
Either way, I think it should have been an error before as well, as we don't finalize the resize as it should be. Changed.